apiserver/facades/client/bundle/bundle.go

```diff
@@ -151,10 +156,111 @@ func (b *BundleAPI) GetChanges(args params.BundleChangesParams) (params.BundleCh
 }
 
 // ExportBundle exports the current model configuration as bundle.
-func (b *BundleAPI) ExportBundle() (params.StringResult, error) {
-	return params.StringResult{}, nil
+func (b *BundleAPI) ExportBundle() (params.BytesResult, error) {
+	if err := b.checkCanRead(); err != nil {
+		return params.BytesResult{}, common.ServerError(common.ErrPerm)
+	}
+
+	var exportConfig state.ExportConfig
+
+	// Skip attributes which are not required.
+	exportConfig.SkipActions = true
+	exportConfig.SkipCloudImageMetadata = true
+	exportConfig.SkipCredentials = true
+	exportConfig.SkipIPAddresses = true
+	exportConfig.SkipSSHHostKeys = true
+	exportConfig.SkipStatusHistory = true
+	exportConfig.SkipLinkLayerDevices = true
+
+	model, err := b.backend.ExportPartial(exportConfig)
+	if err != nil {
+		return params.BytesResult{}, common.ServerError(err)
+	}
+
+	err = model.Validate()
+	if err != nil {
+		return params.BytesResult{}, common.ServerError(err)
+	}
+
+	// Fill it in charm.BundleData datastructure.
+	bundleData, err := b.FillBundleData(model)
+	if err != nil {
+		return params.BytesResult{}, common.ServerError(err)
+	}
+
+	bytes, err := yaml.Marshal(bundleData)
+	if err != nil {
+		return params.BytesResult{}, common.ServerError(err)
+	}
+
+	return params.BytesResult{
+		Result: bytes,
+	}, nil
 }
 
 // Mask the new method from V1 API.
 // ExportBundle is not in V1 API.
 func (u *APIv1) ExportBundle() (_, _ struct{}) { return }
+
+// Fills the bundledata datastructure required for the exportBundle.
+func (b *BundleAPI) FillBundleData(model description.Model) (*charm.BundleData, error) {
+	data := &charm.BundleData{
+		Series:       series.LatestLts(),
+		Applications: make(map[string]*charm.ApplicationSpec),
+		Relations:    [][]string{},
+		Machines:     make(map[string]*charm.MachineSpec),
+	}
+
+	if len(model.Applications()) == 0 {
+		return &charm.BundleData{}, errors.NotFoundf("application")
```

Comment:
Is this going to make sense to a user? They have not requested a particular application.

I think we want to say here that there was nothing to export because this model had no applications... or similar message.